docs/rules-engine/README.md

```diff
@@ -0,0 +1,169 @@
+# Otter Rules Engine
+
+## Introduction
+
+This document describes the various concepts and models that are part of the [Otter rules engine module](https://www.npmjs.com/package/@o3r/rules-engine).
+It serves as an introduction course before you can integrate it within your application following the
+[rules engine implementation guide](./how-to-use/README.md).
+
+As a reminder, the rules engine is a client-based mechanism that evaluates a list of [Rulesets](#ruleset) to define [actions](#action) to
+apply based on the application's context.
+
+## Description
+
+### Ruleset
+
+The ``Ruleset`` object exposes a sorted list of rules to process one after the other.
+
+It may include optional conditions to enable or disable the whole Ruleset based on simple trigger such as the date
+of execution of the Ruleset (validityRange) or the presence of specific components in the page (see 
+[linked components](./how-to-use/dedicated-component-ruleset.md)).<br>
+If any of the rules belonging to the same ruleset fails due to an error, none of the resulting actions from any of these
+rules will be executed as they may be deeply linked.
+
+Rulesets are described by the [ruleset schema](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/rules-engine/schemas/rulesets.schema.json) exposed in the [@o3r/rules-engine](https://www.npmjs.com/package/@o3r/rules-engine) package.
+
+### Rule
+
+A ``Rule`` contains a set of conditions that outputs a list of actions after processing. 
+
+These conditions and actions are referred to as blocks and follow a tree architecture: 
+a block can be an ``IfElseBlock`` (branch of the tree) or an ``ActionBlock`` (leave of the tree).
+
+The ``IfElseBlock`` contains conditions to meet as well as the blocks to execute depending on the success or failure of 
+the condition. 
+The conditions can be linked via ``AND`` and ``OR`` expressions.
+
+An ``IfElseBlock`` can result in another ``IfElseBlock`` or an ``ActionBlock``. 
+
+This allows for nested conditions as demonstrated by the [nested condition Ruleset example](./examples/nested-conditions.md).
+
+> [!NOTE]
+> If no condition is defined in the ``IfElseBlock``, its ``successElements`` will always be executed.
+
+A ``Rule`` also contains information needed to optimize the reevaluation of its conditions:
+* ``inputRuntimeFacts`` : the input runtime facts the rule is based on
+* ``inputFacts`` : the application facts that are used by the rule
+* ``outputRuntimeFacts`` : the runtime facts that are updated/set by the rule
+
+Thanks to these parameters, the rule engines can make sure to only trigger rules evaluation if the input and output of
```

Comment:
```suggestion
Thanks to these parameters, the rule engines can make sure to only trigger rules evaluation if the `inputFacts` of
```
The `outputRuntimeFacts` cannot be the trigger of the rule evaluation, as they depends on `inputFacts` no ?